### ISSUE TITLE
A question about image pre-process

### ISSUE THREAD

**Issue by liuzhengzhe** [devlearning-gt/3DShapeGen#14]
Thanks for your great work! I have a question: I use "best_model_img_3DOF.pth.tar" to generate 3D shape from an image, and how to preprocess the image? I tried image=image/255.0, the result seems not correct. Can you help me? 

Thanks with regards.

![2](https://user-images.githubusercontent.com/8349580/147907111-b5626d1f-03ee-4823-a50a-2f9cf73f87c5.jpg)
The result is: 
![image](https://user-images.githubusercontent.com/8349580/147907125-f5905357-79ff-43c8-ae86-9fbd45272e3e.png)

**Comment by keerthan2** [comment]
@liuzhengzhe can you elaborate on what code did you run to get this output ?.

**Comment by liuzhengzhe** [comment]
Hi, Can you provide a simple script to generate 3D shape from a
single image? Thanks very much.

Zhengzhe

RA Keerthan ***@***.***> 于2022年1月17日周一 17:24写道：

**Comment by keerthan2** [comment]
Well, I'm also stuck in the 3D generation part. The sdf values that I'm getting are not proper. The preprocessing steps mentioned by the author is as below (taken from [dataloader.py](https://github.com/rehg-lab/3DShapeGen/blob/master/SDFNet/dataloader.py)). 

`img_transform = transforms.Compose([
            transforms.Resize(input_size),
            transforms.ToTensor(),
            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
        ])`

You can check my version of a simple test code [here](https://github.com/keerthan2/3DShapeGen/blob/master/SDFNet/test.py) which takes in input a image and outputs sdf values. The sdf values are by itself improper wherein the minimum sdf is positive!. I tried changing the dimensions of the cube, but still no luck. 

@liuzhengzhe can you send the code that you used to get the result that you just posted ?.

**Comment by ngailapdi** [comment]
Hi, thank you for your interest in our work.

Shape models are very sensitive to input images (as shown in section 4.6 of the paper) and that’s why images generated with different rendering processes might cause the model to output low-quality results beyond the training set characteristics.

**Comment by keerthan2** [comment]
Hi @ngailapdi, thank you for the response.

Do you have a test script that produces the 3D mesh from an image ?. The file I referenced in my earlier comment (which is [here](https://github.com/keerthan2/3DShapeGen/blob/master/SDFNet/test.py)) is intended for that purpose, but I'm not getting the desired output. Maybe its because of the variance in rendering/lighting conditions of the image, but to be very sure, I would highly appreciate if you can maybe share a code / instructions on how to test your model to generate 3D mesh from a single image.

**Comment by ngailapdi** [comment]
Hi @keerthan2,

You should be able to directly run `python eval.py` after changing the paths in your `config.py` to point to where your inputs are and specify the `test` field In your json. This should work with any number of test images.

**Comment by keerthan2** [comment]
Hi @ngailapdi,

I already tried that, but the minimum SDF value is still positive (which shouldn't be the case right ?). Since I was trying it out in colab, I got a permission denied error while doing executing `computeMarchingCubes`, which I dint bother to resolve since my SDF values dint seem to make sense in the first place. I used best_model_img_3DOF.pth.tar weight file.

Do you have any idea on what I might be missing or please let me know what additional information you need.

**Comment by liuzhengzhe** [comment]
> Well, I'm also stuck in the 3D generation part. The sdf values that I'm getting are not proper. The preprocessing steps mentioned by the author is as below (taken from [dataloader.py](https://github.com/rehg-lab/3DShapeGen/blob/master/SDFNet/dataloader.py)).
> 
> `img_transform = transforms.Compose([ transforms.Resize(input_size), transforms.ToTensor(), transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]) ])`
> 
> You can check my version of a simple test code [here](https://github.com/keerthan2/3DShapeGen/blob/master/SDFNet/test.py) which takes in input a image and outputs sdf values. The sdf values are by itself improper wherein the minimum sdf is positive!. I tried changing the dimensions of the cube, but still no luck.
> 
> @liuzhengzhe can you send the code that you used to get the result that you just posted ?.

Hi, thanks for your script! The difference of my script is just changing your "transforms.Resize(input_size)," line to  "transforms.Resize([224, 224]),".

**Comment by ngailapdi** [comment]
Hi @keerthan2 

Are you trying to reconstruct from our released rendered images?
